res/controllers/midi-components-0.0.js

```diff
@@ -301,8 +309,48 @@
         shift: function () {
             this.inKey = 'hotcue_' + this.number + '_clear';
         },
+        getColor: function() {
+            if (this.colorIdKey !== undefined) {
+                return color.predefinedColorFromId(engine.getValue(this.group,this.colorIdKey));
+            } else {
+                return false;
+            }
+        },
+        output: function(value) {
+            var outval = this.outValueScale(value);
+            if (this.colorIdKey !== undefined && outval !== this.off) {
+                this.outputColor(engine.getValue(this.group, this.colorIdKey));
```

Comment:
It has to be a separate function because it gets bound to a callback in L348. In C++ this would be a private method but unfortunately there is no such thing is JS.